src/legacy/core_plugins/expressions/public/np_ready/public/render.ts

```diff
@@ -65,22 +66,24 @@ export class ExpressionRenderHandler {
     };
   }
 
-  render = async (data: Data) => {
+  render = (data: Data) => {
```

Comment:
it seems that this was not really needed for lens (having render return promise). Its not something you would even consume if using ExpressionRendererComponent. This is there for the cases where we would consume `render` without the `loader`. So if you get your data in your own manner, you might do `render(data)` where its helpful if this returns a promise and you don't need to subscribe/unsubcribe to observable. No strong feelings about this hwoever, but its something that will start a discussion .

Reply:
I think what you're saying is a small behavior change, but probably a good one: We can return a new Promise which resolves once the `handlers.done` is called by the renderer, or rejects if there is one of the error cases.

Reply:
I think I won't make this change in this PR- if we want render to return a promise, there is probably a consumer of this code other than Lens